asm/vm/asm_vm

```diff
@@ -1340,6 +1567,87 @@ EOF
   retry 3 kpt pkg get --auto-set=false "${KPT_URL}" asm
 }
 
+install_expansion_gateway() {
+  info "Installing the expansion gateway..."
+  retry 5 istioctl install -f "${EXPANSION_GATEWAY_FILE}"
+
+  # Prevent the stderr buffer from ^ messing up the terminal output below
+  sleep 1
+  info "...done!"
+}
+
+expose_istiod() {
+  info "Exposing the control plane for VM workloads..."
+  retry 3 kubectl apply -f "${EXPOSE_ISTIOD_SERVICE}"
+
+  ASM_REVISIONS="$(strip_last_char "${ASM_REVISIONS}")"
+  for rev in $(echo "${ASM_REVISIONS}" | tr ',' '\n'); do
+    kpt cfg set asm anthos.servicemesh.istiodHost "istiod-${rev}.istio-system.svc.cluster.local"
+    kpt cfg set asm anthos.servicemesh.istiod-vs-name "istiod-vs-${rev}"
+    kpt cfg set asm anthos.servicemesh.istiod-dr-name "istiod-dr-${rev}"
+  done
+
+  retry 3 kubectl apply -f "${EXPOSE_ISTIOD_SERVICE}"
+}
+
+install_google_identity_provider() {
+  info "Registering GCE Identity Provider in the cluster..."
+  retry 3 kubectl apply -f asm/identity-provider/identityprovider-crd.yaml
+  retry 3 kubectl apply -f asm/identity-provider/googleidp.yaml
+}
+
+enable_service_mesh_feature() {
+  info "Enabling the service mesh feature..."
+
+  local TOKEN
+  TOKEN="$(retry 2 gcloud \
```

Comment:
unset this variable after it's no longer necessary so we don't have it in a variable longer than necessary

Reply:
done